src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/hmi/vi_get_vehicle_data_response.cc

```diff
@@ -79,6 +79,12 @@ void VIGetVehicleDataResponse::Run() {
   } else {
     event.set_smart_object(*message_);
     policy_handler_.OnVehicleDataUpdated(*message_);
+    if ((*message_).keyExists(strings::msg_params) &&
```

Comment:
Just checking when this is called? Is Core requesting upon launch?

Reply:
@kshala-ford Yes.  You are right. SDL core requests Vehicle Info from the HMI upon launch. And when response is received - SDL checks if VIN code presents.

Reply:
Why was this added? 

Also GetVehicleData response will be called anytime there is a get vehicle data request from an app. Not just at startup.

Reply:
Also is this vin code sent to the webengine applications?

Reply:
@JackLivio 
>Why was this added?

_In this place SDL gets unique vin code, which is used to create WebEngine device. It is always unique for every vehicle._

>Also GetVehicleData response will be called anytime there is a get vehicle data request from an app. Not just at startup.

_Yes, but only at startup HMI sends vin code. Even if consider wrong case when HMI sends the VIN code many times - it should be the same one. SDL checks if device based on this VIN code does exist- and does nothing if yes._

>Also is this vin code sent to the webengine applications?

_No, this vin code is used in connection handler to create WebEngine device_

Reply:
I think tieing the webengine feature to a vehicle data rpc & parameter is an unnecessary dependency. If the VIN isnt being used in the connection handshake at all then it does not matter if the ID is unique or not. This also means the HMI does not need to supply such a value, there can be a default string unique to the websocket server built into core's code instead.

Reply:
I want to make a suggestion (not knowing the complexity): Could it be possible to create a device upon a SetAppProperties request using a random device identifier? Alternatively a MAC address of an ethernet device can be used.

Reply:
I am ok with either the ivi's MAC or a randomly generated device string.

Reply:
Hi @JackLivio 
I have  one question regarding your comment here:
https://github.com/smartdevicelink/sdl_core/pull/3221#discussion_r377769153

I have several ideas how to avoid dependency on VIN code and VehicleInfo
I'd like to avoid redundant time spending on implementation.
That's why I would like to ask you which one of the proposed solutions is 
suitable for you? 

1) We can use unique machine-id (just by reading /etc/machine-id file on Linux)
   The machine ID is usually generated from a random source during
   system installation and stays constant for all subsequent boots.
   
   http://man7.org/linux/man-pages/man5/machine-id.5.html
   
   Benefits: It is unique and is the same across ign cycles and resets. 
   Drawbacks: The way of getting unique machine id may be different on different OS
   
2) We can use sum of all MAC ADDRESSES as a base for hash function
   to generate unique device ID for WebEngine device

   Benefits: It is unique and is the same across ign cycles and resets but UNTIL 
   network adapters or interfaces are changed. MAC Addresses will be changed - thus     device id will be different. 
   
   Drawbacks: The way of getting MAC address is different on different OS
   
3) SDL can generate unique device ID and write it internally to the file system.
   E.g. to the file app_info.dat which is already present and used for resumption.
   
   Benefits: It is unique and doesn't depend on OS at all
   
   Drawbacks: The same device ID will be kept until MASTER RESET, as MASTER RESET
   resets the SDL to its initial state(thus clears all files created)
   
 So what it the most suitable approach by your opinion?

Reply:
@AByzhynar Thank you for these suggestions. I prefer option 3 because it does not have an OS dependency.

Reply:
@AByzhynar do you have an update here?  We would like to be able to review and merge this PR by the end of day tomorrow (February 26), so testing on Core 6.1 can begin.  Thanks!

Reply:
@JackLivio Fixed in 55adcffc7139ac8768673bafc6d379d3efa44da2